app/src/main/java/com/example/ecostyle/view/CheckoutFragment.kt

```diff
@@ -42,28 +48,30 @@ class CheckoutFragment : Fragment() {
         })
         recyclerView.adapter = cartAdapter
 
-        checkoutButton = view.findViewById(R.id.checkout_button)
-        checkoutButton.isEnabled = false
-
-        loadCartItems()
-
-        // Verificar stock y proceder al pago
+        // Comportamiento del botón de checkout
         checkoutButton.setOnClickListener {
-            verifyStockBeforeCheckout()
+            if (isNetworkAvailable(requireContext())) {
+                verifyStockBeforeCheckout()
+            } else {
+                Toast.makeText(context, "\n" +
+                        "Payment cannot be made offline. Please try again later.", Toast.LENGTH_LONG).show()
```

Comment:
```suggestion
                        "Payment cannot be made offline. Please try again later.", Toast.LENGTH_SHORT).show()
```
creo que deberiamos tener los toast cortos, a veces se sobrelapan